support/ci/cargo_update.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail 
+ 
+# shellcheck source=./support/ci/shared.sh 
+source ./support/ci/shared.sh 
+
+branch="ci/cargo-update-$(date +"%Y%m%d")"
```

Comment:
Might be worth making this more unique to avoid collisions
 ```suggestion
branch="ci/cargo-update-$(date +"%Y%m%d")-$RANDOM"
```

Reply:
> Since this is an automated job running once a week, we're not going to have to worry about collisions in the general case. However, if we're talking about re-running a potentially failed invocation of this workload, then we will have to think about collisions.

Given our experience elsewhere, I was worried about collisions. I think making the time more specific would help, but also think there's little harm in doing both.